packages/ckeditor5-table/src/tableediting.js

```diff
@@ -159,3 +163,38 @@ export default class TableEditing extends Plugin {
 		return [ TableUtils ];
 	}
 }
+
+// Creates a mapper callback to adjust model position mappings in a table cell containing a paragraph, which is bound to its parent
+// (to the table cell). Only positions after this paragraph have to be adjusted, because after binding this paragraph to the table cell,
+// elements located after this paragraph would point either to a non-existent offset inside `tableCell` (if paragraph is empty), or after
+// the first character of the paragraph's text. See https://github.com/ckeditor/ckeditor5/issues/10116.
+//
+// <tableCell><paragraph></paragraph>^</tableCell> -> <td>^&nbsp;</td>
+//
+// <tableCell><paragraph>foobar</paragraph>^</tableCell> -> <td>foobar^</td>
+//
+// @param {module:engine/view/view~View} editingView
+// @returns {Function}
+function mapTableCellModelPositionToView( editingView ) {
+	return ( evt, data ) => {
+		const modelParent = data.modelPosition.parent;
+		const modelNodeBefore = data.modelPosition.nodeBefore;
+
+		if ( !modelParent.is( 'element', 'tableCell' ) ) {
+			return;
+		}
+
+		if ( !modelNodeBefore || !modelNodeBefore.is( 'element', 'paragraph' ) ) {
+			return;
+		}
+
+		const viewNodeBefore = data.mapper.toViewElement( modelNodeBefore );
+		const viewParent = data.mapper.toViewElement( modelParent );
+
+		if ( viewNodeBefore === viewParent ) {
+			// Since the paragraph has already been bound to its parent, update the current position in the model with paragraph's
+			// max offset, so it points to the place which should normally (in all other cases) be the end position of this paragraph.
+			data.viewPosition = editingView.createPositionAt( viewParent, modelNodeBefore.maxOffset );
```

Comment:
You should use :

```javascript
data.viewPosition = data.mapper.findPositionIn( viewParent, modelNodeBefore.maxOffset );
```

To find a proper offset in the view element. The current implementation works by accident (by the fact that [https://github.com/ckeditor/ckeditor5/blob/8805829e23ddd010d1f3852f98cfc5c08e4d776f/packages/ckeditor5-engine/src/view/element.js#L653](https://github.com/ckeditor/ckeditor5/blob/8805829e23ddd010d1f3852f98cfc5c08e4d776f/packages/ckeditor5-engine/src/view/element.js#L653) is appending the new child to the list) of children.

Also, this is reordering those elements but I guess it's not in the scope of this PR?

Reply:
Thanks, I've changed it as you suggested.

Tasks related to the reordering of elements are already reported (https://github.com/ckeditor/ckeditor5/issues/10118, https://github.com/ckeditor/ckeditor5/issues/10119). This PR is only to fix the editor crash when there was a comment in the empty table cell.